EasyPost.Tests/BetaFeaturesTests/ParametersTests/ParametersTest.cs

```diff
@@ -0,0 +1,152 @@
+                Street2 = "Apt 20",
+                City = "San Francisco",
+                State = "CA",
+                Zip = "94107",
+                Country = "US",
+            };
+
+            var dictionary = parameters.ToDictionary();
+
+            // Check that a dictionary was created correctly
+            Assert.NotNull(dictionary);
+
+            // Check that the dictionary contains "address" + default "verify" and "verify_strict" top-level keys
+            Assert.True(dictionary.ContainsKey("address"));
+            Assert.True(dictionary.ContainsKey("verify"));
+            Assert.True(dictionary.ContainsKey("verify_strict"));
+
+            // Check that the "address" sub-dictionary was created correctly
+            var addressData = dictionary["address"] as Dictionary<string, object>;
+            Assert.NotNull(addressData);
+
+            // Check that the "address" sub-dictionary contains the 6 elements we set + default "residential" boolean
+            Assert.True(addressData.ContainsKey("street1"));
+            Assert.True(addressData.ContainsKey("street2"));
+            Assert.True(addressData.ContainsKey("city"));
+            Assert.True(addressData.ContainsKey("state"));
+            Assert.True(addressData.ContainsKey("zip"));
+            Assert.True(addressData.ContainsKey("country"));
+            Assert.True(addressData.ContainsKey("residential"));
+
+            // Check that the "street1" key was set with the correct value
+            Assert.NotNull(addressData["street1"]);
+            Assert.Equal(street, addressData["street1"]);
+        }
+
+        [Fact]
+        [Testing.Parameters]
+        public void TestParametersToDictionaryWithSubDictionary()
+        {
+            const string streetA = "388 Townsend St";
+            const string streetB = "123 Main St";
+
+            Address preExistingAddressObject = new Address
+            {
+                Street1 = streetA,
+            };
+
+            Addresses.Create newAddressCreationParameters = new Addresses.Create
+            {
+                Street1 = streetB,
+            };
+
+            // Users can pass in an existing Address object as the "ToAddress" parameter
+            var parameters = new Shipments.Create
+            {
+                IsReturn = false,
+                ToAddress = preExistingAddressObject,
+            };
+
+            // No errors here confirm that the dictionary was serialized to the expected schema
+            var dictionary = parameters.ToDictionary();
+            var shipmentData = dictionary["shipment"] as Dictionary<string, object>;
+            var addressData = shipmentData["to_address"] as Dictionary<string, object>;
+
+            // The value of "street1" should be the value of "streetA" via the Address object
+            Assert.Equal(streetA, addressData["street1"]);
+
+            // Users can also pass in an Address.Create parameter object as the "ToAddress" parameter
```

Comment:
Plural right?

I don't know that I love this interface of plural? You'd only ever be creating one of anything. What happened to the `Params` name? Weren't we going to do something like `AddressParams` or `AddressCreateParams`?

```suggestion
            // Users can also pass in an Addresses.Create parameter object as the "ToAddress" parameter
```

Reply:
The namespace to the object (the full name of the class) is `EasyPost.BetaFeatures.Parameters.Addresses.Create`, so I figured it was pretty self-explanatory that these are parameters.

As for the plural, it's more for categorization. There would be `Addresses.All`, `Addresses.Retrieve`, `Addresses.Create`, etc.

Reply:
Ah, with them missing the full namespace it was confusing. This is probably fine.